controller/controller_server/utils.py

```diff
@@ -201,6 +236,35 @@ def generate_csi_create_snapshot_response(new_snapshot, source_volume_id):
     return res
 
 
+def _get_supported_capabilities(volume_capabilities):
+    if volume_capabilities.HasField(config.VOLUME_CAPABILITIES_FIELD_ACCESS_TYPE_MOUNT):
+        return csi_pb2.VolumeCapability(
+            mount=csi_pb2.VolumeCapability.MountVolume(fs_type=volume_capabilities.mount.fs_type),
+            access_mode=csi_pb2.VolumeCapability.AccessMode(mode=volume_capabilities.access_mode.mode))
+
+    elif volume_capabilities.HasField(config.VOLUME_CAPABILITIES_FIELD_ACCESS_TYPE_BLOCK):
```

Comment:
if I understand correctly, if e.g. a 3rd access type will be added in the future, the current code will return `confirmed`.
I know that `the CO SHALL reconcile successful capability-validation responses by comparing the validated capabilities with those that it had originally requested.` should overcome this problem, but we can still easily avoid the confirmation in some cases, like said example.

Reply:
a. there is already validation in validate_csi_volume_capabilities.
the check on the volume is missing. I will add